### PR TITLE
What makes entando a highly available architecture

### PR DESCRIPTION
**Pull request by lward27**

### CODE REVIEW

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2022/12/scalingkubernetesstraws.jpg
+---
+
+High availability within the software world means a user can employ a technology to do a piece of work when needed. For software to be *highly available*, it needs to be **Scalable, Resilient, Reliable, and Durable**. To understand what makes an Application Composition Platform like Entando a Highly Available Architecture, we must understand what makes Kubernetes highly available.
+
+# An Analogy
+
+If you ever drink a Caprisun Cooler or a kid's Juice box, they have these tiny straws that require extra effort to draw the liquid up. Compare this to drinking through a straw from Mcdonald's or a giant smoothie straw: Less effort for more volume. This same idea is how *vertical* scalability works. Increase the size of the machine, and theoretically, throughput increases.
+
+Increase the size of the machine, and theoretically, throughput increases.
```
Comment (dsharpe) [inline posts/what-makes-entando-a-highly-avialable-architecture.md:19]: Did you mean to repeat this sentence?

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- kubernetes
+- devops
+- docker
+title: "What Makes Entando a Highly Available Architecture?"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2022/12/scalingkubernetesstraws.jpg
+---
+
+High availability within the software world means a user can employ a technology to do a piece of work when needed. For software to be *highly available*, it needs to be **Scalable, Resilient, Reliable, and Durable**. To understand what makes an Application Composition Platform like Entando a Highly Available Architecture, we must understand what makes Kubernetes highly available.
+
+# An Analogy
```
Comment (dsharpe) [inline posts/what-makes-entando-a-highly-avialable-architecture.md:15]: Love the analogy! :)

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+- Initiating a backup save of the database.
+- Initiating a backup restore on the new node.
+
+# Many Masters
+
+All this talk about scaling nodes and refreshing pods begs the question, what manages these resources? The answer is the Kubernetes "Master Node," also known as the "Primary Node." Scaling the worker nodes provides Kubernetes with more resources. More worker nodes bring more CPU and memory resources for pod scheduling. What about scaling the Primary Node itself? There are many benefits to scaling the Primary Node in addition to the worker nodes.
+
+Configuring Kubernetes to have multiple Primary Nodes will bulletproof your application deployment. As mentioned before, Entando allows you to do this and supports it with minimal configuration. The recommended minimum number of Primary Nodes for a highly available Kubernetes deployment is 3.
+
+Having multiple masters also improves the network performance of the cluster because the multiple masters act together as a unified data center.  This all serves to reduce the possible failure of the etcd datastore, API server, controller manager and scheduler, and all of the worker nodes.  In a single Master Node setup
```
Comment (dsharpe) [inline posts/what-makes-entando-a-highly-avialable-architecture.md:77]: I think you did not finish your last sentence :)

**Review by dsharpe — CHANGES_REQUESTED** [review]

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Caching techniques can also help scale database reads. Storing frequently accessed data in a separate cache keeps the load off of the database.  When a user or an application goes to access the data, it may not even hit the database, pulling from the cache instead.
+
+A third way to scale database reads is to implement database sharding. This strategy is also the primary method used to scale database writes.
+
+### Scaling Database Writes
+
+A database write happens when a user or program wishes to store a piece of data on the database. Since every database replica will also need to contain this piece of data, increasing replicas does not fix the issue as it does for database reads. Enter what is known as "database sharding."
+
+Sharding a database is the process of spreading the data across multiple databases. Spreading the data out can be as simple as putting records that start with the character 'A' through 'L' in one database and the remaining 'M' to 'Z' records in a different database. This arrangement means we need another component in front of the database to handle the routing. Some databases support sharding natively, and others will require a third-party tool.
```
Comment (cjin-1) [inline posts/what-makes-entando-a-highly-avialable-architecture.md:62]: Is this term coined by Entando - or is this an often-used term across tech?

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Configuring Kubernetes to have multiple Primary Nodes will bulletproof your application deployment. As mentioned before, Entando allows you to do this and supports it with minimal configuration. The recommended minimum number of Primary Nodes for a highly available Kubernetes deployment is 3.
+
+Having multiple masters also improves the network performance of the cluster because the multiple masters act together as a unified data center.  This all serves to reduce the possible failure of the etcd datastore, API server, controller manager and scheduler, and all of the worker nodes.  In a single Master Node setup
+
+# Conclusion
+
+Building application's on Kubernetes makes scaling easy. A cutting-edge platform like Entando ensures that your critical applications scale when and how they need to. If you want to learn more about Entando, check out the website [here](https://entando.com/). If you need help implementing Kubernetes or Entando in your organization, drop us a line at [contact@ippon.tech](mailto:contact@ippon.tech).
```
Comment (cjin-1) [inline posts/what-makes-entando-a-highly-avialable-architecture.md:81]: Building _applications_

**Review by cjin-1 — CHANGES_REQUESTED** [review]

File: posts/what-makes-entando-a-highly-avialable-architecture.md (hunk 2 above)
Comment (pkrish123) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:15]: Same! This is a great one!

**Review by pkrish123 — CHANGES_REQUESTED** [review]

File: posts/what-makes-entando-a-highly-avialable-architecture.md (hunk 1 above)
Comment (lward27) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:19]: Nope! I'll fix this :)

File: posts/what-makes-entando-a-highly-avialable-architecture.md (hunk 2 above)
Comment (lward27) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:15]: Thanks y'all!

File: posts/what-makes-entando-a-highly-avialable-architecture.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+# An Analogy
+
+If you ever drink a Caprisun Cooler or a kid's Juice box, they have these tiny straws that require extra effort to draw the liquid up. Compare this to drinking through a straw from Mcdonald's or a giant smoothie straw: Less effort for more volume. This same idea is how *vertical* scalability works. Increase the size of the machine, and theoretically, throughput increases.
+
+Increase the size of the machine, and theoretically, throughput increases.
+What would happen if the big straw were to crack, however? It would become even more difficult to suck the liquid through. Instead of a wide straw, let's use two, three, or even ten straws! Having multiple straws would still mean high throughput. And, if one of the straws were to crack, plenty of beverage would still flow through the other straws. A new straw could replace the broken straw without stopping the *slurp!*
+
+This multiple-straw analogy is the central concept behind "horizontal" scalability. Kubernetes is a highly scalable, resilient, and reliable architecture. Entando is built on Kubernetes and shares all of these excellent benefits.
```
[the remark this answers is not in the document]
Comment (lward27) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:22]: Thank's Chris!

File: posts/what-makes-entando-a-highly-avialable-architecture.md (hunk 3 above)
Comment (lward27) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:77]: Whoops!

File: posts/what-makes-entando-a-highly-avialable-architecture.md (hunk 4 above)
Comment (lward27) [reply posts/what-makes-entando-a-highly-avialable-architecture.md:62]: Sharding is a well-known term in Database land.

**Review by pkrish123 — APPROVED** [review]

**Review by dsharpe — APPROVED** [review]

**Pull request merged**